### ISSUE TITLE
Returning 0 results for "some" query

### ISSUE THREAD

**Issue by romanoff** [blevesearch/bleve#1548]
Following code reproduces problem:

```golang
package main

import (
	"github.com/blevesearch/bleve"
	"fmt"
)

type ArticleData struct {
	Name string
	Content string
}

func main() {
	mapping := bleve.NewIndexMapping()
	index, err := bleve.NewMemOnly(mapping)
	if err != nil {
		fmt.Println(err)
		return
	}
	index.Index("1", ArticleData{Name: "Some", Content: "some some some some text"})
	query := bleve.NewMatchPhraseQuery("some")
	search := bleve.NewSearchRequest(query)	
	searchResult, err := index.Search(search)
	fmt.Println(searchResult)
	fmt.Println(err)
}

```
Tried different other queries instead of `bleve.NewMatchPhraseQuery`, but they worked same way. If I replace `some` with `text`, it finds `text` right away. Not sure why `some` text has this problem

**Comment by Jxic** [comment]
I got the same problem here but for the value "a" as well
```
mapping := bleve.NewIndexMapping()
index, err := bleve.New("test1", mapping)
if err != nil {
	fmt.Println("creating index err: ", err)
	return
}

data := testStruct{
	Name:   "a",
	FieldB: "a",
}

index.Index(data.Name, data)

query := bleve.NewMatchQuery("a")
search := bleve.NewSearchRequest(query)
searchResult, err := index.Search(search)
if err != nil {
	fmt.Println("searching err: ", err)
	return
}
fmt.Println(searchResult.Hits.Len())
fmt.Println(searchResult)
fmt.Println("search done")
```
returns ```No matches```

the version I'm using is v2.0.2

**Comment by Jxic** [comment]
I found another issue https://github.com/blevesearch/bleve/issues/1424, which describes the solution to this problem

**Comment by mschoch** [comment]
@romanoff sorry for the late reply somehow I missed this issue.  @Jxic has it right, the issue is that the word "some" is an English stop word, which our default analyzer removes:

https://github.com/blevesearch/bleve/blob/ae28975038cb25655da968e3f043210749ba382b/analysis/lang/en/stop_words_en.go#L276

The linked issue #1424 has an example of how to define an analyzer like "en" that keeps stop words in the index.

**Comment by Jxic** [comment]
> @romanoff sorry for the late reply somehow I missed this issue. @Jxic has it right, the issue is that the word "some" is an English stop word, which our default analyzer removes:
> 
> https://github.com/blevesearch/bleve/blob/ae28975038cb25655da968e3f043210749ba382b/analysis/lang/en/stop_words_en.go#L276
> 
> The linked issue #1424 has an example of how to define an analyzer like "en" that keeps stop words in the index.

Hi, there's one thing I'm still unsure of. That is when I have text like ```EN```,  the search does not find anything, but as I change it to ```en```, the search works just fine. However, I don't spot any stop word like ```EN``` in the ```stop_words_en.go``` file, and I am using the default mapping with standard analyzer. Any suggestion on the cause?

**Comment by mschoch** [comment]
@Jxic can you provide a complete example for this case?  It's not clear to me if "EN" is in the document or the search.

**Comment by Jxic** [comment]
@mschoch Sure, here is a full example

```go
package main

import (
	"fmt"

	"github.com/blevesearch/bleve/v2"
)

func main() {
	testIndex()
}

type testStruct struct {
	Name   string
	FieldB string
}

func testIndex() {
	mapping := bleve.NewIndexMapping()
	index, err := bleve.New("test1", mapping)
	if err != nil {
		fmt.Println("creating index err: ", err)
		return
	}

	sName := "EN" // works fine when I change this variable to "en"
	data := testStruct{
		Name:   sName,
		FieldB: "TB2rj",
	}

	index.Index(data.Name, data)

	query := bleve.NewTermQuery(sName)
	query.SetField("Name")

	search := bleve.NewSearchRequest(query)
	searchResult, err := index.Search(search)
	if err != nil {
		fmt.Println("searching err: ", err)
		return
	}
	fmt.Println(searchResult.Hits.Len())
	fmt.Println(searchResult)
	fmt.Println("search done")
}

```
The above code would return ```No matches```

**Comment by Jxic** [comment]
Oh, maybe it's because the ```lowercase``` token filter changes the input and I actually have to search for text in lowercase?

**Comment by mschoch** [comment]
@Jxic yes that is the explanation. The TermQuery always look for the exact term, without performing any analysis.  This is useful for exact matches (things like IDs) and for debugging which terms are really there.  In this case, if you switch to MatchQuery, you will find a match, because the search term is analyzed the same way as the indexed term.

**Comment by blaubaer** [comment]
@mschoch Hi, I have the same problem. I really have a case where the `bleve.NewTermQuery` makes sense to me. But how I can ensure that if there are matching results they are returned and not like now (mostly a 50:50 chance) that sometimes nothing is returned?

**Comment by mschoch** [comment]
@blaubaer I do not understand the question. We do not have any cases of unexplained missing results.  If you can provide a reproducible example of some behavior, we will investigate it.

**Comment by blaubaer** [comment]
@mschoch You're right. I missed up the context a little bit. I'm currently on it to create a smaller version of our code which produces with a chance of 50% an empty result. Once I'm done I post it here, sorry for the confusion.

**Comment by mschoch** [comment]
@blaubaer please create a new issue.
